### PR TITLE
Add a section on hermit-sync

### PR DESCRIPTION
**Pull request by mkroening**

### CODE REVIEW

File: content/this-month/2022-11/index.md [hunk 1]
```diff
@@ -113,6 +113,24 @@ The initial nightlies containing the prebuilt UEFI targets revealed some issues
 Finally, we [changed the C compiler for the UEFI targets from gcc to clang](https://github.com/rust-lang/rust/pull/104622), which resolved some linker problems. 
 As of the 2022-11-22 nightly, the three UEFI targets should be fully usable, which means you no longer need to use the unstable `-Zbuild-std` feature.
 
+### [`hermitcore/hermit-sync`](https://github.com/hermitcore/hermit-sync)
+<span class="maintainers">(Section written by [@mkroening](https://github.com/mkroening))</span>
+
+While working on the [libhermit-rs] kernel, I moved the synchronization primitives targeted at kernel development into a separate crate: [hermit-sync]. Whilst I was there, I also enhanced the code and made it independent from libhermit, so the code can now be used in other projects as well. (At least x86_64 and aarch64 ones for the moment). The crate contains the following features:
```
Comment (phil-opp) [inline content/this-month/2022-11/index.md:119]: Minor formatting nits:

```suggestion
While working on the [`libhermit-rs`] kernel, I moved the synchronization primitives targeted at kernel development into a separate crate: [`hermit-sync`]. Whilst I was there, I also enhanced the code and made it independent from libhermit, so the code can now be used in other projects as well. (At least x86_64 and aarch64 ones for the moment). The crate contains the following features:
```

File: content/this-month/2022-11/index.md [hunk 2]
```diff
@@ -113,6 +113,24 @@ The initial nightlies containing the prebuilt UEFI targets revealed some issues
...
+
+* a function for running a closure without interrupts
+* a simple spinlock with exponential backoff
+* a fair ticket lock with exponential backoff
+* `OnceCell` and `Lazy` based on [generic_once_cell] (another recent project of mine)
```
Comment (phil-opp) [inline content/this-month/2022-11/index.md:124]:
```suggestion
* `OnceCell` and `Lazy` based on [`generic_once_cell`] (another recent project of mine)
```

File: content/this-month/2022-11/index.md [hunk 3]
```diff
@@ -113,6 +113,24 @@ The initial nightlies containing the prebuilt UEFI targets revealed some issues
...
+
+All of these primitives are available with and without interrupt safety.
```
Comment (phil-opp) [inline content/this-month/2022-11/index.md:126]: I think a link to docs.rs would be useful, e.g. here:

```suggestion
All of these primitives are available with and without interrupt safety. See the [API docs](https://docs.rs/hermit-sync/latest/hermit_sync/) for more details.
```

**Review by phil-opp — APPROVED** [review]
Cool project, thanks for submitting!

**Comment by mkroening** [comment]
Thanks! I applied your suggestions and a few more. :)

**Review by phil-opp — APPROVED** [review]
Thanks!

**Pull request merged**